### PR TITLE
News

### PR DESCRIPTION
**Pull request by Marcosnr97**
## Summary
Added design to the news fragment.
Recycler view implementation to handle the news list.
News list mock to visualize the list behaviour
Added SwipeRefresh behaviour to news list

## Screenshots
Peek 2021-04-22 13-46
Peek 2021-04-22 13-49

### CODE REVIEW

File: app/src/main/java/ar/com/wolox/android/example/model/News.kt [hunk 1]
```diff
@@ -0,0 +1,179 @@
+package ar.com.wolox.android.example.model
+
+import android.content.res.Resources
+import androidx.annotation.DrawableRes
+import ar.com.wolox.android.R
+
+data class News(
```
Comment (ffasitella) [inline app/src/main/java/ar/com/wolox/android/example/model/News.kt:7]: This is a model, its name shouldn't be plural 😅

File: app/src/main/java/ar/com/wolox/android/example/ui/home/news/NewsAdapter.kt [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import androidx.recyclerview.widget.RecyclerView
+import ar.com.wolox.android.R
+import ar.com.wolox.android.example.model.News
+
+class NewsAdapter(private val news: List<News>) : RecyclerView.Adapter<NewsViewHolder>() {
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): NewsViewHolder {
+        val view = LayoutInflater.from(parent.context)
+                .inflate(R.layout.view_new_item, parent, false)
```
Comment (ffasitella) [inline app/src/main/java/ar/com/wolox/android/example/ui/home/news/NewsAdapter.kt]: Change `attachToRoot` to true

File: app/src/main/java/ar/com/wolox/android/example/ui/home/news/NewsAdapter.kt [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    }
+
+    override fun getItemCount(): Int = news.size
+
+    override fun onBindViewHolder(holder: NewsViewHolder, position: Int) {
+        with(news.elementAt(position)) {
+            holder.newsTitle.text = title
+            holder.newsDescription.text = description
+            if (isLiked) holder.likeButton.setImageResource(R.drawable.ic_like_on)
+            else holder.likeButton.setImageResource(R.drawable.ic_like_off)
```
Comment (ffasitella) [inline app/src/main/java/ar/com/wolox/android/example/ui/home/news/NewsAdapter.kt:23]: You can implement this the same way you implemented the tab icons. ;)

File: app/src/main/java/ar/com/wolox/android/example/ui/home/news/NewsFragment.kt [hunk 4]
```diff
@@ -10,6 +14,27 @@ class NewsFragment @Inject constructor() : WolmoFragment<FragmentNewsBinding, Ne
     override fun layout() = R.layout.fragment_news
 
     override fun init() {
+        with(binding) {
+            recyclerView.layoutManager = LinearLayoutManager(context)
+            val newsAdapter = NewsAdapter(newsList(resources))
+            recyclerView.adapter = newsAdapter
+            swipeRefresh.setColorSchemeResources(R.color.colorAccent)
```
Comment (ffasitella) [inline app/src/main/java/ar/com/wolox/android/example/ui/home/news/NewsFragment.kt:21]: Can you implement this in the XML?

File: app/src/main/java/ar/com/wolox/android/example/ui/home/news/NewsFragment.kt [hunk 5]
```diff
@@ -10,6 +14,27 @@ class NewsFragment @Inject constructor() : WolmoFragment<FragmentNewsBinding, Ne
...
+                recyclerView.visibility = View.GONE
+                noNewsMessage.visibility = View.VISIBLE
+            }
+        }
+    }
+
+    override fun setListeners() {
+        with(binding) {
+            swipeRefresh.setOnRefreshListener(OnRefreshListener {
+                swipeRefresh.isRefreshing = false
```
Comment (ffasitella) [inline app/src/main/java/ar/com/wolox/android/example/ui/home/news/NewsFragment.kt]: This should be a decission made in the presenter. The fragment just has the responsibility to hide and show the refreshing icon

File: app/src/main/java/ar/com/wolox/android/example/ui/home/news/NewsViewHolder.kt [hunk 6]
```diff
@@ -0,0 +1,13 @@
...
+import android.view.View
+import android.widget.ImageView
+import android.widget.TextView
+import androidx.recyclerview.widget.RecyclerView
+import ar.com.wolox.android.R
+
+class NewsViewHolder(view: View) : RecyclerView.ViewHolder(view) {
+    val newsTitle: TextView = view.findViewById(R.id.newsTitle)
+    val newsDescription: TextView = view.findViewById(R.id.newsDescription)
+    val likeButton: ImageView = view.findViewById(R.id.likeButton)
```
Comment (ffasitella) [inline app/src/main/java/ar/com/wolox/android/example/ui/home/news/NewsViewHolder.kt:13]: You can implement a classic method called `populate()` to set the value of these views

**Review by ffasitella — CHANGES_REQUESTED** [review]
Good job! Just some issues to solve before merging

File: app/src/main/java/ar/com/wolox/android/example/ui/home/news/NewsFragment.kt (hunk 4 above)
Comment (Marcosnr97) [reply app/src/main/java/ar/com/wolox/android/example/ui/home/news/NewsFragment.kt:21]: Nope, swipeRefresh color setting is only available outside the XML

**Review by jcruz-wolox — APPROVED** [review]
Good job, approved for me

**Review by ffasitella — APPROVED** [review]
🚀

**Pull request merged**